cpp/src/arrow/pretty_print.cc

```diff
@@ -197,7 +197,9 @@ class ArrayPrinter : public PrettyPrinter {
   }
 
   template <typename T>
-  inline typename std::enable_if<std::is_base_of<ListArray, T>::value, Status>::type
+  inline typename std::enable_if<std::is_base_of<ListArray, T>::value ||
+                                     std::is_base_of<LargeListArray, T>::value,
```

Comment:
Perhaps a `BaseListArray` base class would be useful (and perhaps also a `BaseListType`).
Also `enable_if_list<T>` in `type_traits.h` may deserve updating.